announcement/models/announcement.py

```diff
@@ -84,6 +85,17 @@ def _compute_allowed_user_ids(self):
             announcement.allowed_user_ids = announcement.user_group_ids.users
             announcement.allowed_users_count = len(announcement.user_group_ids.users)
 
+    @api.onchange("is_general_announcement")
+    def _onchange_is_general_announcement(self):
```

Comment:
Instead of an onchange, you should convert `user_group_ids` to a computed writable field for not depending on the UI for making the "internal" change.

Reply:
But I have a doubt about this, in the compute method I must limit myself to calculate the value of the user_group_ids field, but to set the rest of the values, for example "self.specific_user_ids = False" I must keep the onchange, right? or would it be correct to include everything in the compute method?

Reply:
It's not mandatory to assign the value for all the records:

```python
@api.depends("is_general_announcement")
def _compute_group_user_ids:
    for record in self:
        if record.is_general_announcement:
            record.group_user_ids = [(6, 0, self.env.ref("base.group_user").ids)]
```

Now, if you uncheck the mark, do you want to keep it the same group or not. I don't have an strong opinion on that.

Reply:
The behaviour should be as follows. If you tick the option "General announcement" you assign the group "Internal user" to the field "user_groups_id" but hide the field to select the announcement type which by default is selected the announcement type for specific user. If the "General announcement" option is unchecked, the field for selecting the announcement type reappears and both the "user_groups_id" and "specific_user_ids" fields must be clear.